pages/electronics-library-dev.tsx

```diff
@@ -109,6 +86,59 @@ const Home = ({
   const [activeProduct, setActiveProduct] =
     useState<IElectronicsComponent | null>(null);
 
+  const [searchField, setSearchField] = useState<string>("");
+  const [searchCategoryTags, setSearchCategoryTags] = useState<CategoryTags[]>([CategoryTags.ShowAll]);
+  const onSearchChange = (e: React.FormEvent<HTMLInputElement>) => {
+    setSearchField(e.currentTarget.value);
+  };
+  const onTagSelect = (e: React.FormEvent<HTMLInputElement>) => {
+    setSearchCategoryTags([e.currentTarget.value as CategoryTags])
+  };
+
+  const [filteredComponents,setFilteredComponents] = useState<IElectronicsComponent[]>(electronicComponents)
+  const filterComponents = () => {
+    let componentList:IElectronicsComponent[] = electronicComponents;
+    if (searchCategoryTags[0] !== CategoryTags.ShowAll) {
+      componentList = componentList.filter((component) => {
+        return component.category?component.category.toLowerCase().includes(searchCategoryTags[0].toLowerCase()):null;
+      });
+    }
+
+    if(searchField) {      
+      const searcher = new Searcher(componentList,{keySelector: (obj:IElectronicsComponent) => [obj.category, obj.name, obj.shortDescription, obj.description, obj.tags?.join(" "), obj.shownTo, obj.leadTime]});
+      setFilteredComponents(searcher.search(searchField));
+    } else {
+      setFilteredComponents(componentList);
+    }
+  }
+
+  const [resultsHeader, setResultsHeader] = useState<JSX.Element>();
+  const buildResultsHeader = () => {
+    setResultsHeader(
+      <>
+        {`${searchField===''?'Showing':'Searching all'}`}{' '}
+        <span style={{color: categoryObject[searchCategoryTags[0]].color, fontWeight: 700}}>
+          {categoryObject[searchCategoryTags[0]].result}
+        </span>
+        <span style={{transition:"0.2s all ease", opacity: searchField?'1':'0'}}>
+          {' '}with "{searchField}"
+        </span>
```

Comment:
Good call. I originally had that animation in before the debouncer since the text updating while typing was distracting. I put in the fix you recommended. Thanks!